sanity-check.sh

```diff
@@ -14,19 +15,27 @@ printf "\n\nGenerating Makefile files...\n"
 printf "\n\nGenerating and building C files...\n"
 make -C ${OUT_DIR} || { echo "Error building C files" ; exit 1 ; }
 
-printf "\n\nDownloading module schema...\n"
+printf "\n\nDownloading JSON Schema schema..\n"
+(cd ${OUT_DIR} && wget --no-cache ${JSON_SCHEMA_URL}) \
+    || { echo "Error getting json schema base" ; exit 1 ; }
+
+printf "\n\nDownloading and validating module schema...\n"
 (cd ${OUT_DIR} && wget --no-cache ${URL}/${MODULE_SCHEMA_FILE}) \
     || { echo "Error getting module json-schema" ; exit 1 ; }
+jsonschema -i ${OUT_DIR}/${MODULE_SCHEMA_FILE} ${OUT_DIR}/schema \
+           || { echo "Invalid JSON schema" ; exit 1 ; }
 
-printf "\n\nDownloading general info schema...\n"
+printf "\n\nDownloading and validating general info schema...\n"
 (cd ${OUT_DIR} && wget --no-cache ${URL}/${GENERAL_SCHEMA_FILE}) \
     || { echo "Error getting general json-schema" ; exit 1 ; }
+jsonschema -i ${OUT_DIR}/${GENERAL_SCHEMA_FILE} ${OUT_DIR}/schema \
+           || { echo "Invalid JSON schema" ; exit 1 ; }
```

Comment:
This error message could be more useful if it contained information if the error is from module or general file

Reply:
it will be just below  "Downloading and validating XXXX schema" :)